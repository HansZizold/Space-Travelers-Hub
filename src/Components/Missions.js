import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { joinMission } from '../redux/missions/missions';

const Missions = () => {
  const missions = useSelector((state) => state.missions);
  const dispatch = useDispatch();
  return (
    <div className="missions_container">
      <table>
        <tbody>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th />
          </tr>
          {missions &&  missions.map((mission) => (
          <tr key={mission.mission_id}>
            <td className="missionName">{mission.mission_name}</td>
            <td className="missionDescription">{mission.description}</td>
            <td className={`userBadge ${mission.reserved ? 'member' : ' '}`}><label>
              {mission.reserved ? 'Active member' : 'NOT A MEMBER'}</label></td>
            <td className="joinButton">
              <button onClick={() => dispatch(joinMission(mission.mission_id))} className={`missionBtn ${mission.reserved ? 'joined' : ' '}`}>
              {mission.reserved ? 'Leave Mission' : 'Join Mission'}
              </button>
            </td>
          </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
}
export default Missions;
