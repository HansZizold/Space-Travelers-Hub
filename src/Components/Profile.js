import React from 'react';

const Profile = () => {
  return (
    <div>ERROR!! Page Not Found</div>
  );
}
export default Profile;
