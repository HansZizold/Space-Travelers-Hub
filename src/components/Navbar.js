import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../images/planet.png'

const Navbar = () => {
  const navItem = [
    {
      id: "1",
      title: "Rockets",
      link: "/",
    },
    {
      id: "2",
      title: "Missions",
      link: "/missions",
    },
    {
      id: "3",
      title: "My Profile",
      link: "/profile"
    }
];
<div className="">
<NavLink to={navItem[0].link}>
  <img src={planet} className="App-logo" alt="logo" />
</NavLink>
  <ul className="showMenu">
    {navItem.map((item) => (
      <li key={item.id}>
        <NavLink
          to={item.link}
        >
          {item.title}
        </NavLink>
      </li>
    ))}
  </ul>
</div>
}

export default Navbar;
