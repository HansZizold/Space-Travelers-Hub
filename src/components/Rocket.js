import React from 'react';
import { useDispatch } from 'react-redux';

import PropTypes from 'prop-types';

import { reserveRocket, cancelRocket } from '../redux/rockets/rockets';

import styles from './Rocket.module.css';

function Rocket(props) {
  const dispatch = useDispatch();
  const {
    id, name, description, image, reserved,
  } = props;
  console.log(reserved);

  const reservation = (elem) => {
    dispatch(reserveRocket(elem.target.id));
  }

  const cancel = (elem) => {
    dispatch(cancelRocket(elem.target.id));
    // console.log(elem.target.id);
  }

  return (
    <div className={styles.rocket}>
      <img src={image} alt={name} height="250" width="300"></img>
      <div>
        <h2>{name}</h2>
        <p>{reserved && <mark>Reserved</mark>}        {description}</p>
        {reserved
          ? <button id={id} type='button' className={styles.cancel} onClick={cancel}>Cancel Reservation</button>
          : <button id={id} type='button' className={styles.reserve} onClick={reservation}>Reserve Rocket</button>
        }
      </div>
    </div>
  );
}

Rocket.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
};
export default Rocket;
