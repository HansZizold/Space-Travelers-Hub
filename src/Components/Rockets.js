import React from 'react';
import { useSelector } from 'react-redux';

import Rocket from './Rocket';

import styles from './Rockets.module.css';

const Rockets = () => {
  const rocketlist = useSelector(state => state.rockets);

  return (
    <div className={styles.books}>
      <div className={styles.booklist}>
        {rocketlist.map((elem) => {
          const {
            id, name, description, image, reserved
          } = elem;
          return (
            <Rocket
              key={id}
              id={id}
              name={name}
              description={description}
              image={image}
              reserved={reserved}
            />
          );
        })}
      </div>
    </div>
  );
}
export default Rockets;
