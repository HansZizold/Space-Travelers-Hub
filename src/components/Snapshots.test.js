import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import store from '../redux/configureStore';
import Rocket from './Rocket';
import Missions from './Missions.js';
import Rockets from './Rockets.js';
import Profile from './Profile.js';
import Navbar from './Navbar';

const renderReservedRockets = (reserved) => {
  const { tree } = render(
    <Provider store={store}>
      <Rocket
        reserved={reserved}
        id={1}
        name="Falcon 1"
        description="Test description"
        image="https://imgur.com/DaCfMsj.jpg"
      />
    </Provider>,
  );
  return tree;
};

const renderJoinedMissions = (joined) => {
  const { tree } = render(
    <Provider store={store}>
      <Missions
        joined={joined}
        id={1}
        name="Thaicom"
        description="Test description"
      />
    </Provider>,
  );
  return tree;
};

describe('Rocket and Mission component tests', () => {
  it('Renders first Rocket', () => {
    expect(renderReservedRockets(true)).toMatchSnapshot();
  });
  it('Renders first Mission', () => {
    expect(renderJoinedMissions(true)).toMatchSnapshot();
  });
});

describe('Renders main Components', () => {
  it('Renders Navbar component', () => {
    const tree = render(
      <Router>
        <Navbar />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Renders Rockets component', () => {
    const tree = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Renders Missions component', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Renders Profile component', () => {
    const tree = render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
