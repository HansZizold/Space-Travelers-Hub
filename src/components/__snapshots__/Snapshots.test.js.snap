// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders main Components Renders Missions component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="missions_container"
      >
        <table>
          <tbody>
            <tr>
              <th>
                Mission
              </th>
              <th>
                Description
              </th>
              <th>
                Status
              </th>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions_container"
    >
      <table>
        <tbody>
          <tr>
            <th>
              Mission
            </th>
            <th>
              Description
            </th>
            <th>
              Status
            </th>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders main Components Renders Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar"
      >
        <div
          class="logotitle"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <img
              alt="logo"
              class="App-logo"
              src="planet.png"
            />
          </a>
          <h2>
            Space Travelers Hub
          </h2>
        </div>
        <ul
          class="menu"
        >
          <li>
            <a
              aria-current="page"
              class="navitem active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class="navitem"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class="navitem"
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar"
    >
      <div
        class="logotitle"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <img
            alt="logo"
            class="App-logo"
            src="planet.png"
          />
        </a>
        <h2>
          Space Travelers Hub
        </h2>
      </div>
      <ul
        class="menu"
      >
        <li>
          <a
            aria-current="page"
            class="navitem active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class="navitem"
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class="navitem"
            href="/profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders main Components Renders Profile component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="profile"
      >
        <div
          class="missions"
        >
          <h2>
            My Missions
          </h2>
          <ul
            class="list"
          />
        </div>
        <div
          class="rockets"
        >
          <h2>
            My Rockets
          </h2>
          <ul
            class="list"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="profile"
    >
      <div
        class="missions"
      >
        <h2>
          My Missions
        </h2>
        <ul
          class="list"
        />
      </div>
      <div
        class="rockets"
      >
        <h2>
          My Rockets
        </h2>
        <ul
          class="list"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders main Components Renders Rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rocket and Mission component tests Renders first Mission 1`] = `undefined`;

exports[`Rocket and Mission component tests Renders first Rocket 1`] = `undefined`;
